src/odemis/gui/cont/acquisition/cryo_acq.py

```diff
@@ -198,7 +233,23 @@ def _on_acquisition(self, is_acquiring: bool):
         # update the layout
         self._panel.Layout()
 
-    def _on_acquire(self, _):
+    def _get_acqui_streams(self, evt: wx.Event) -> List[stream.Stream]:
+        """
+        Get the acquisition streams based on the acqui mode,
+        button pressed, and the current view
+        """
+        acq_streams = self._acquiStreams.value
+        sender = evt.GetEventObject()
+        if (self.acqui_mode is guimod.AcquiMode.FIBSEM and
+                sender == self._panel.btn_cryosecom_acquire):
+            if self._tab_data.is_sem_active_view:
+                acq_streams = [s for s in acq_streams if isinstance(s, SEMStream)]
+            if self._tab_data.is_fib_active_view:
+                acq_streams = [s for s in acq_streams if isinstance(s, FIBStream)]
```

Comment:
Instead of such ad-hoc check, assuming the views have defined classes, it should be possible to make it generic like:
```
view = self._tab_data.focussedView.value
acq_streams = [s for s in acq_streams if view.is_compatible(s.__class__)]
```